### DISCUSSION TITLE
How to tell if a notebook is running in JupyterLite or regular JupyterLab

### DISCUSSION THREAD

**Discussion by giswqs** [jupyterlite/jupyterlite#875] [Q&A] [answered]
Since `pandas.read_csv(http_url)` does not work with pyodide (https://github.com/jupyterlite/jupyterlite/issues/119), I would like to rewrite some functions of my package so that the functions can automatically use pyodide's `fetch()` function to retrieve files. The problem is that how can we figure out whether a notebook is running within JupyterLite or a regular JupyterLab? Thanks.

**Comment by jobovy** [comment, marked as answer]
Take a look at this: https://pyodide.org/en/stable/usage/faq.html#how-to-detect-that-code-is-run-with-pyodide

**Reply by giswqs** [reply]
Awesome! This is exactly what I need. Thank you. 

```python
import sys
if "pyodide" in sys.modules:
    # do sth
```
